### DISCUSSION TITLE
[Question]: How Do I prevent the Assistant changing the user's input when calling an action?

### DISCUSSION THREAD

**Discussion by dividor** [danny-avila/LibreChat#2976] [Q&A] [answered]
### What is your question?

I am using an OpenAI assistant which has an action configured to access long-term memory. I would like what the user entered to be passed to the action, but looking at the action debug in the Librechat UI, I see it calls the action and summarizes the user's input. This breaks the memory retrieval.

Is there a way to just send exactly what the user entered or at least have some control over the LLM which is changing the intent please?

### More Details

.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment, marked as answer]
You would have to hack the code a bit to force the user's input as the input for your desired action, every time.

Otherwise, the only other way is through very specific prompting, because OpenAI Assistants will always form their own inputs for tools (but can always be overriden programmatically)

**Reply by dividor** [reply]
OK, thanks!
